Applications/Spire/Source/Login/LoginWindow.cpp

```diff
@@ -101,22 +114,17 @@ LoginWindow::LoginWindow(const std::string& version, QWidget* parent)
   content_layout->addWidget(m_status_label);
   content_layout->setStretchFactor(m_status_label, 14);
   content_layout->addStretch(20);
-  m_username_line_edit = new QLineEdit(this);
-  connect(m_username_line_edit, &QLineEdit::textEdited,
-    [=] {on_input_updated();});
-  m_username_line_edit->setPlaceholderText(tr("Username"));
-  m_username_line_edit->setSizePolicy(QSizePolicy::Expanding,
+  m_username_text_box = new TextBox(this);
+  m_username_text_box->get_model()->connect_current_signal(
```

Comment:
Is the scoped connection needed here because of the model?

Reply:
Correct.